torchdynamo/symbolic_convert.py

```diff
@@ -1125,6 +1123,12 @@ def inline_call_(parent, func, args, kwargs):
             )
 
         try:
+            # TODO(before land): I didn't research this in detail, need to check
+            # what else we need to handle here
+            if func.has_self():
+                # remove self
+                assert args[0] is args[1]
+                args = args[1:]
```

Comment:
There are two values of `self`:
1) `self` at compile time
2) `self` at call time

I think this will incorrectly use the first at call time, which may be incorrect.

Reply:
just to make sure I understand your comment correctly,
1. current code hackily removes self to ensure the number of arguments is right
2. you are pointing out that there are other things we need to handle, such as value of self during call time not matching value of self during compile time (which my current code doesn't do anything about)

is that a correct understanding?

if yes, would implementing your suggestion below result in a the method call having `is_self` equal to False, because it would be unpacked?